### PR TITLE
New Features & Updates

### PR DESCRIPTION
**Pull request by smit2444**
- Developed new "Start Collaboration" component: this allows user to create new collaboration with multiple collaborators
- Developed new "start_collaboration" endpoint
- Modified "Search" component
- Modified "Invitations" endpoints (Send, Accept, Reject, Withdrawal & Cancel)
- Working on "Collaboration Details" component

### CODE REVIEW

File: web_application/Backend/FlaskApp/app.py [hunk 1]
```diff
@@ -431,34 +430,55 @@ def send_invitation():
...
+            return jsonify({'message': 'Invitation already sent for this collaboration'}), 200
 
         invitation = {
             'uuid': str(uuid.uuid4()),
             'receiver_id': ObjectId(receiver_id),
             'sender_id': ObjectId(sender_id),
             'status': 'pending',
-            'phenotype': phenotype
+            'phenotype': phenotype,
+            'collaboration_uuid': collaboration_uuid,
```
Comment (Kevincj) [inline web_application/Backend/FlaskApp/app.py]:
```suggestion
            'collaboration_uuid': collaboration_id,
```

File: web_application/Backend/FlaskApp/app.py [hunk 2]
```diff
@@ -431,34 +430,55 @@ def send_invitation():
         data = request.get_json()
         receiver_id = data.get('receiver_id')
         phenotype = data.get('phenotype')
+        collaboration_uuid = data.get('collaboration_uuid')
```
Comment (Kevincj) [inline web_application/Backend/FlaskApp/app.py]:
```suggestion
        collaboration_id = data.get('collaboration_id')
```

Comment (Kevincj) [reply web_application/Backend/FlaskApp/app.py]: Let's use xxx_id everywhere. UUID would be confusing if we use `receiver_id` or something at the same time.

File: web_application/Backend/FlaskApp/app.py [hunk 3]
```diff
@@ -509,7 +535,40 @@ def withdraw_invitation():
...
+            db.collaborations.update_one(
+                {'uuid': collaboration_uuid, 'invited_users.user_id': receiver_id},
+                {'$set': {'invited_users.$.status': 'withdrawn'}}
+            )
+
+        return jsonify({'message': 'Invitation and collaboration status updated successfully to withdrawn'}), 200
+    else:
+        return jsonify({'message': 'No matching invitation found'}), 404
+    
+@app.route('/api/cancelinvitation', methods=['POST'])
```
Comment (Kevincj) [inline web_application/Backend/FlaskApp/app.py]:
```suggestion
@app.route('/api/cancel_invitation', methods=['POST'])
```

File: web_application/Backend/FlaskApp/app.py [hunk 4]
```diff
@@ -532,10 +591,127 @@ def reject_invitation():
...
+
+        if not collab_name:
+            logging.error("Collaboration name missing in the request")
+            return jsonify({"error": "Collaboration name is required"}), 400
+
+        collaboration = {
+            'uuid': str(uuid.uuid4()),
+            'name': collab_name,
+            'experiments': experiments,
+            'raw_data': raw_data,
```
Comment (Kevincj) [inline web_application/Backend/FlaskApp/app.py:631]: Should we put data and metadata inside 'invited_users' and make it per 'user'(collaborator)?

**Comment by Kevincj** [comment]
Great work!

File: web_application/Backend/FlaskApp/app.py (hunk 2 above)
Comment (smit2444) [reply web_application/Backend/FlaskApp/app.py]: Note.

**Pull request merged**